web-registry/src/pages/MyEcocredits.tsx

```diff
@@ -19,24 +45,117 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
 }));
 
-export const MyEcocredits: React.FC = () => {
-  const theme = useTheme();
+const WrappedMyEcocredits: React.FC<WithBasketsProps> = ({ baskets }) => {
   const styles = useStyles();
-  const walletContext = useWallet();
-  const accountAddress = walletContext.wallet?.address;
+  const theme = useTheme();
+
+  const handleTxQueued = (): void => {
+    setIsProcessingModalOpen(true);
+  };
+
+  const handleTxSuccessfulModalClose = (): void => {
+    setCardItems(undefined);
+    setIsTxSuccessfulModalTitle(undefined);
+    setDeliverTxResponse(undefined);
+  };
+
+  const handleTxDelivered = (): void => {
+    setIsProcessingModalOpen(false);
+    // Refetch basket/ecocredits data so it shows latest values
+    fetchBasketTokens();
+    fetchCredits();
+  };
+
+  // TODO handle error when signing and broadcasting tx
+  const { signAndBroadcast, setDeliverTxResponse, wallet, deliverTxResponse } =
+    useMsgClient(handleTxQueued, handleTxDelivered);
+  const accountAddress = wallet?.address;
+  const { credits, fetchCredits } = useEcocredits(accountAddress);
+  const { basketTokens, fetchBasketTokens } = useBasketTokens(
+    accountAddress,
+    baskets,
+  );
+  const basketsWithClasses = useBasketsWithClasses(baskets);
+  const { api } = useLedger();
+
+  const [creditBaskets, setCreditBaskets] = useState<
+    (QueryBasketResponse | undefined)[][]
+  >([]);
+  const [basketPutOpen, setBasketPutOpen] = useState<number>(-1);
+  const [isProcessingModalOpen, setIsProcessingModalOpen] = useState(false);
+  const [selectedBasketDenom, setSelectedBasketDenom] = useState('');
+  const [isTxSuccessfulModalTitle, setIsTxSuccessfulModalTitle] = useState<
+    string | undefined
+  >(undefined);
+  const [cardItems, setCardItems] = useState<Item[] | undefined>(undefined);
+
+  useEffect(() => {
+    // Get available baskets to put credits into
+    if (basketsWithClasses && basketsWithClasses.length > 0) {
+      setCreditBaskets(
+        credits.map(c =>
+          basketsWithClasses.filter(b => b?.classes.includes(c.class_id)),
+        ),
+      );
+    }
+  }, [credits, basketsWithClasses]);
+
+  const openTakeModal = (rowIndex: number): void => {
+    if (!accountAddress) return;
+    const selectedBasket = basketsWithClasses?.[rowIndex]?.basket;
+    if (selectedBasket?.basketDenom) {
+      setSelectedBasketDenom(selectedBasket.basketDenom);
+    }
+  };
+
+  const handleTakeCredits = async (values: MsgTakeValues): Promise<void> => {
+    const msgClient = api?.msgClient;
```

Comment:
close the take modal here (see submit for put)

Reply:
done